mlir/lib/Transforms/Utils/DialectConversion.cpp

```diff
@@ -1992,6 +2020,8 @@ void ConversionPatternRewriterImpl::replaceAllUsesWith(
   replacedValues.insert(from);
 #endif // NDEBUG
 
+  assert(!functor &&
```

Comment:
I feel this might be worth using `llvm::report_fatal_error` for to make sure users are not misusing the API, even in release mode